LiskHQ/lisk-desktop
# Implement new blocks and transactions buttons - Closes #2510 #2513

### What issue have I solved?

Resolves #2510
Resolves #2513

I decided to handle both tickets in one PR because once the button is implemented for Blocks, adding the button for Transactions is just 10 lines in https://github.com/LiskHQ/lisk-desktop/blob/2510-implement-load-new-blocks-button/src/components/shared/transactionsTable/index.js#L81-L90 so a separate PR would IMO be overkill.

### How have I implemented/fixed it?

- [x] Create `useServiceSocketUpdates` hook for listening to new blocks
- [x] Create the `LoadLatestButton` shared component that uses `useServiceSocketUpdates` hook
- [x] Implement styling of the `LoadLatestButton` component to match [design](https://projects.invisionapp.com/d/main#/console/18906983/396861464/preview)
- [x] Use `LoadLatestButton` on Blocks monitor 
- [x] Use `LoadLatestButton` on Transactions monitor 

Note that there are the following deliberate divergencies from design:
- Font-size of the button is 15px, like any other Primary button of Size L. (The design has 13px.)
- The button is placed between the box header and table header. (In design, it is between table header and table content.)
 I think this way it looks fine and the code is cleaner as I don't have to pass the button inside the toolbox Table component.

### How has this been tested?

- Go to Block Monitor
- Wait for blocks to be forged so that the button appears
- Click the button to get the new block displayed

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-desktop/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-desktop/blob/development/docs/CSS_GUIDE.md)
